src/python/pants/init/util.py

```diff
@@ -27,12 +30,33 @@ def init_workdir(global_options: OptionValueContainer) -> str:
     workdir_base = global_options.pants_physical_workdir_base
     workdir_dst = os.path.join(workdir_base, safe_filename_from_path(workdir_src))
 
+    def maybe_create_source_control_symlink():
+        """If the buildroot we are working in has a source control administrative directory, we need
+        to add a link to it from the physical workdir we create.
+
+        Tools which run from the physical workdir expect to be able to glean repository information
+        directly. For instance node_binary targets are built using ad-hoc build scripts which can do
+        this.
+        """
+        # Don't link anything unless this option is enabled.
+        if not global_options.pants_physical_workdir_source_control:
+            return
+        for scm_state_dir in SOURCE_CONTROL_DIRS:
+            scm_source_path = os.path.join(get_buildroot(), scm_state_dir)
+            scm_target_path = os.path.join(workdir_dst, scm_state_dir)
```

Comment:
This is a bit strange... the `.git` directory will end up "inside of" the workdir?

Reply:
Well yeah a link to it will be there. I get that it seems strange, but it isn't much different than having the workdir be a child of something with a .git directory. I am open to other ideas about how to fix this, but I liked this solution because it doesn't seem to open up any new holes in the workdir. It just replaces on that is missing when using the symlinked workdir.